test_runner/src/main/kotlin/ftl/shard/TestMethodDuration.kt

```diff
@@ -0,0 +1,35 @@
+package ftl.shard
+
+import ftl.args.AndroidArgs
+import ftl.args.IArgs
+import ftl.reports.xml.model.JUnitTestCase
+import ftl.reports.xml.model.JUnitTestResult
+
+fun createTestMethodDurationMap(junitResult: JUnitTestResult, args: IArgs): Map<String, Double> {
+    // Create a map with information from previous junit run
+    return mutableMapOf<String, Double>().apply {
+        junitResult.testsuites?.forEach { testsuite ->
+            testsuite.testcases
+                ?.asSequence()
+                ?.filter { testcase -> !testcase.empty() && testcase.time != null }
+                ?.map { testcase ->
+                    val key = if (args is AndroidArgs) testcase.androidKey() else testcase.iosKey()
+                    val time = testcase.time!!.toDouble()
+                    key to time
+                }
+                ?.filter { (_, time) -> time >= 0 }
+                ?.forEach { (key, time) -> this[key] = time }
```

Comment:
This new version adds 5 additional unnecessary null checks and 5 lines of code more. Also I am not sure if this filters and maps are necessary if at last we still have forEach and additions to map. IMO it is not simpler than previous version. Pls rollback this change, and I will approve pull request

Reply:
reverted. 
What I have in my mind is that previously there was also null checks and filters and maps change the code sequential way